iothub_client/inc/iothub_client_core_common.h

```diff
@@ -17,137 +17,138 @@ extern "C"
 {
 #endif
 
-#define IOTHUB_CLIENT_FILE_UPLOAD_RESULT_VALUES \
-    FILE_UPLOAD_OK, \
-    FILE_UPLOAD_ERROR
+typedef enum IOTHUB_CLIENT_FILE_UPLOAD_RESULT_TAG
+{
+    FILE_UPLOAD_OK = 0,
```

Comment:
just as a general comment...not to be fixed in this PR because it would take a while. But if and after this gets merged we should go through these enums and properly document them with doxygen. EG:
`/**<  */`

Reply:
Agree.  Let's track - [https://github.com/Azure/azure-iot-sdk-c/issues/1354](https://github.com/Azure/azure-iot-sdk-c/issues/1354).

---
In reply to: [350323583](https://github.com/Azure/azure-iot-sdk-c/pull/1342#discussion_r350323583) [](ancestors = 350323583)